integrations/outlook/src/client.ts

```diff
@@ -0,0 +1,116 @@
+import moment from 'moment'
+import { Client, ResponseType } from '@microsoft/microsoft-graph-client'
+import { TokenCredentialAuthenticationProvider } from '@microsoft/microsoft-graph-client/authProviders/azureTokenCredentials'
+import { ClientSecretCredential } from '@azure/identity'
+import type { AckFunction, IntegrationContext } from '@botpress/sdk'
+import type { Conversation, Message } from '@botpress/client'
+
+type SendMessageProps = {
+  ctx: IntegrationContext
+  conversation: Conversation
+  message: Message
+  ack: AckFunction
+  body: any
+}
+
+interface LifecycleEvent {
+  subscriptionId: string
+  subscriptionExpirationDateTime: string
+  tenantId: string
+  clientState: string
+  lifecycleEvent: 'subscriptionRemoved' | 'missed' | 'reauthorizationRequired'
+}
+
+export class GraphApi {
+  private client: Client
+  constructor(tenantId: string, clientId: string, clientSecret: string) {
+    const credential = new ClientSecretCredential(tenantId, clientId, clientSecret)
+    const authProvider = new TokenCredentialAuthenticationProvider(credential, {
+      scopes: ['https://graph.microsoft.com/.default'],
+    })
+
+    this.client = Client.initWithMiddleware({
+      //debugLogging: true,
+      authProvider,
+    })
+  }
+
+  /**
+   * Send Mail
+   */
+  public sendMail = async ({ message, conversation, ctx, ack, body }: SendMessageProps): Promise<void> => {
+    const refMessageId = conversation.tags['outlook:refMessageId']
+
+    if (!refMessageId) {
+      console.log('conv tag missing: outlook:refMessageId')
+      return
+    }
+
+    try {
+      await this.client
+        .api(`/users/${ctx.configuration.emailAddress}/messages/${refMessageId}/replyAll`)
+        .responseType(ResponseType.RAW)
+        .post({
+          message: {
+            body,
+          },
+        })
+      ack({ tags: { 'outlook:id': `${message.id}` } })
+    } catch (error) {
+      console.log(error.message)
+    }
+
+    return
+  }
+
+  /**
+   * Subscribe webhook
+   */
+  public subscribeWebhook = async (webhookUrl: string, ctx: IntegrationContext): Promise<string> => {
+    const expirationDateTime = this.generateExpirationDate()
```

Comment:
**question:** I don't get this. Will the integration stop working after 20 minutes ? Is it possible not to set an expiration date ?

Reply:
Ok, I just saw your comment above. I answered it right under.